From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Axel Garcia K." <git@axelgk.com>
Date: Tue, 5 May 2026 18:57:44 +0000
Subject: [PATCH] Add SF2000 QuickNES alignment guards

Align the static video buffer and avoid unaligned field access patterns on SF2000 so the core can run under the MIPS32 alignment constraints.
---
 libretro/libretro.cpp   | 6 +++++-
 nes_emu/blargg_common.h | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/libretro/libretro.cpp b/libretro/libretro.cpp
index b9afa9dbff0356b14c79a26abe4cdedf5d66cf3a..8cc47fd9f2ea5fcaeb1b8acd77b5848983bcf2ca 100644
--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -1330,7 +1330,11 @@ bool retro_load_game(const struct retro_game_info *info)
    static uint8_t video_buffer[Nes_Emu::image_width * (Nes_Emu::image_height + 16)];
    emu->set_pixels(video_buffer + (8 * Nes_Emu::image_width), Nes_Emu::image_width);
 #else
-   static uint8_t video_buffer[videoBufferWidth * videoBufferHeight];
+   static uint8_t video_buffer[videoBufferWidth * videoBufferHeight]
+#if defined(SF2000)
+      __attribute__((aligned(16)))
+#endif
+      ;
    emu->set_pixels(video_buffer, videoBufferWidth);
 #endif
 
diff --git a/nes_emu/blargg_common.h b/nes_emu/blargg_common.h
index 413fd6b19b18bdda172844a34324174a44a9c69e..b07c196509edf5215b1d0f402cf29b2b7d8cbe0a 100644
--- a/nes_emu/blargg_common.h
+++ b/nes_emu/blargg_common.h
@@ -52,7 +52,7 @@ arithmetic on smaller types. */
 #endif
 
 typedef struct
-#ifdef NO_UNALIGNED_ACCESS
+#if defined(NO_UNALIGNED_ACCESS) || defined(SF2000)
 __attribute ((packed))
 #endif
 {

base-commit: 7848e1ac22b1c69d056ae4cb57710651ff1dd169
-- 
2.54.0

